### PR TITLE
NOT FOR MERGE: Bionic32

### PR DESCRIPTION
**Pull request by mattip**
I don't have permissions to push a new branch or create a new repo, so am issuing this as a PR. It provides a bionic 32 bit docker image with python 2.7, 3.4, 3.5, 3.6, 3.7, 3.8, and builds a 2.7.15 narrow build

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -23,12 +23,14 @@ RUN echo '#!/bin/sh' > /usr/sbin/policy-rc.d \
 	\
 	&& echo 'Acquire::GzipIndexes "true"; Acquire::CompressionTypes::Order:: "gz";' > /etc/apt/apt.conf.d/docker-gzip-indexes
 
+RUN apt update
+RUN apt install -y gnupg
```
Comment (mattip) [inline Dockerfile:27]: needed for updating apt keys to pull from deadsnakes

File: build_install_pythons.sh [hunk 2]
```diff
@@ -1,25 +1,24 @@
...
-echo "deb http://ppa.launchpad.net/deadsnakes/ppa/ubuntu trusty main" > /etc/apt/sources.list.d/deadsnakes.list
+echo "deb http://ppa.launchpad.net/deadsnakes/ppa/ubuntu bionic main" > /etc/apt/sources.list.d/deadsnakes.list
 apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 6A755776
 apt-get update
 apt-get install -y wget
 PIP_ROOT_URL="https://bootstrap.pypa.io"
 wget $PIP_ROOT_URL/get-pip.py
-for pyver in 3.4 3.5 3.6 2.7 2.6 3.3 ; do
+apt-get install -y python3-distutils
+for pyver in 3.5 3.6 3.7 3.8 2.7; do
```
Comment (mattip) [inline build_install_pythons.sh:12]: It turns out deadsnakes has everything we need. Dropping 3.4, 3.3. Do any projects still need them?

File: build_install_pythons.sh [hunk 3]
```diff
@@ -76,27 +75,6 @@ function build_openssl {
     rm -rf ${froot} ${ftgz}
 }
 
-build_openssl 1.0.2o
```
Comment (mattip) [inline build_install_pythons.sh:79]: wow, that is an old version

File: build_install_pythons.sh [hunk 4]
```diff
@@ -76,27 +75,6 @@ function build_openssl {
...
-pip install certifi
-CERTIFI_CERT=$(python -c "import certifi; print(certifi.where())")
-DEFAULT_CERT=$(python -c"import ssl; print(ssl.get_default_verify_paths().openssl_cafile)")
-echo "CERTIFI_CERT=$CERTIFI_CERT"
-echo "DEFAULT_CERT=$DEFAULT_CERT"
-cp ${CERTIFI_CERT} ${DEFAULT_CERT}
-deactivate
-# Remove virtualenv files from image
-rm -rf venv
-
```
Comment (mattip) [inline build_install_pythons.sh:99]: no longer needed (I think)

**Review by mattip — COMMENTED** [review]
I tested that the image builds but not beyond that, some functional testing would be good

**Comment by mattip** [comment]
I guess the testing comes from multibuild, so if you commit this as a new branch I can add a few tests to the multibuild matrix. I can try that out locally I guess, maybe I will have some time over the weekend. It would be nice to get this going and drop trusty, since it is [officially EOL](https://endoflife.software/operating-systems/linux/ubuntu) (although it has HWE status till April 2022. Should we be more conservative and only upgrade to xenial (16..04)?

**Comment by mattip** [comment]
> I can try that out locally I guess, maybe I will have some time over the weekend

No, it is really hard to replicate the CI run locally. multibuild wants to `docker pull` the image, among other things.

**Comment by mattip** [comment]
If you open a bionic-64 branch I can submit a PR for that too

File: build_install_pythons.sh (hunk 2 above)
Comment (matthew-brett) [reply build_install_pythons.sh:12]: I don't think so.  Multibuild doesn't support those versions, as you know.

**Comment by matthew-brett** [comment]
Matti - maybe time to move this to an org?  I think you can merge pull requests, but otherwise we need an org to give you more access.

**Comment by mattip** [comment]
Sure. What org would you suggest? I don't know if NumPy wants to take this over, maybe a multibuild org? Or we could call it wheelbuilders, and try to get other groups like [cibuildwheel](https://github.com/joerick/cibuildwheel) involved as well :)

**Comment by mattip** [comment]
Closing. https://github.com/multi-build/docker-images already builds 64-bit xenial images, and the future of 32-bit builds for x86 is bleak. Ubuntu has already dropped them.

**Pull request closed without merging**